Fixing issue with sketches missing a Topic name (i.e.: Getting Started)

ubidefeo opened the pull request:
- handling empty `_shadowTopicIn`
- preventing `_syncStatus` from being set to value other than `SYNC_STATUS_SYNCHRONIZED` when empty shadow topic

lxrobotics commented:
Fixed bracing style myself + moved initialisation of topic names with sensible default values into Constructor (where member variables typically should be initialized anyway).

ubidefeo commented:
hi @lxrobotics 
I had fixed the style last night but forgot to push.
good that you also added initialisation in constructor 👍 

I'll do squash + merge and bump the release up.
would you mind merging these changes into __simplify-debug-message-with-variadic-arguments__ then?
